staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -96,9 +96,20 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 			return
 		}
 
-		// TODO(mikedanese): verify the response audience matches one of apiAuds if
-		// non-empty
-
+		if len(apiAuds) > 0 && len(resp.Audiences) > 0 {
```

Comment:
How about something like this?

https://github.com/kubernetes/kubernetes/blob/7e906ae145f1c8d9306c24b95c239550a921531a/pkg/registry/authentication/tokenreview/storage.go#L99